### PR TITLE
Add Has Many Through relation

### PR DESCRIPTION
**Pull request by pietrocaselani**
Now we have support to has many with a third table!

### CODE REVIEW

File: Pod/Classes/GDGEntityManager.h [hunk 1]
```diff
@@ -38,6 +38,8 @@
 
 - (void)hasMany:(NSString *)relationName config:(void (^)(GDGHasManyRelation *))tap;
 
+- (void)hasManyThrough:(NSString *)relationName config:(void (^)(GDGHasManyThroughRelation *))tap;
```
Comment (frlwolf) [inline Pod/Classes/GDGEntityManager.h]: Not saying that this way is wrong, or anything, but I was imagining something like `-hasMany:through:config:`

File: Pod/Classes/GDGEntityQuery.m [hunk 2]
```diff
@@ -106,7 +106,7 @@ - (instancetype)initWithSource:(__kindof GDGSource *)source
 #undef ORDER_BLOCK
 
 		_id = ^GDGEntityQuery *(NSInteger id) {
-			return self.where(^(GDGCondition *condition) {
+			return weakSelf.where(^(GDGCondition *condition) {
```
Comment (frlwolf) [inline Pod/Classes/GDGEntityQuery.m:109]: My bad

File: Pod/Classes/GDGHasManyThroughRelation.m [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+	for (GDGEntity *entity in entities)
+	{
+		NSMutableArray *relatedEntities = idRelationDictionary[@([entity id])];
+
+		[entity setValue:[NSArray arrayWithArray:relatedEntities] forKey:self.name];
+	}
+}
+
+- (void)set:(__kindof NSObject *)value onEntity:(GDGEntity *)entity {}
```
Comment (frlwolf) [inline Pod/Classes/GDGHasManyThroughRelation.m]: Here should be creating the registers in the `n <----> n` source table

**Comment by pietrocaselani** [comment]
- Change all conditions from `NSString*` to `GDGCondition*`;
- Implements `^groupBy` and `^having` on `GDGQuery*`;

**Comment by pietrocaselani** [comment]
:metal: updated!

**Comment by frlwolf** [comment]
There is yet somethings I think could have been done better, but once there is nothing _blocking_ in this pull request, I'll merge it and generate a version. 

Then I'll merge pull request #9.

And finally I'll create a third pull request with some purposed additions.

**Pull request merged**